src/components/rooms-tab.js

```diff
@@ -127,11 +213,35 @@ function RoomsTab({ rooms }) {
       <section className="RoomsTab-section">
         <h2>{t('ownedRoomsHeader')}</h2>
         <Table dataSource={ownedRoomsRows} columns={ownedRoomsColumns} size="middle" />
+        <Restricted to={permissions.CREATE_ROOMS}>
+          <Button className="RoomsTab-newRoomButton" type="primary" shape="circle" icon={<PlusOutlined />} size="large" onClick={handleNewRoomClick} />
+        </Restricted>
       </section>
       <section className="RoomsTab-section">
         <h2>{t('joinedRoomsHeader')}</h2>
         <Table dataSource={membershipRoomsRows} columns={membershipRoomsColumns} size="middle" />
       </section>
+
+      <Modal
```

Comment:
I would like it if we start splitting these modals into their own components, these files where we define them are getting too  long. What do you guys think?

Reply:
I think I would not not extract these, feels like there would be alot of props to pass down to them.

However maybe we can have model mappers for the pages/components that have tables, and extract the mapping between the DB data and the table data (columns and rows).

Reply:
I think if you would in this example split the Modal + its form into its own component, every  state property and event handler for the form can go into the modal. Then the rooms tab becomes the tab + the tables + a button that opens the modal. 
We have the same situation on creating docs. We don't need to do it now, I was just wanting to bring it up.

Reply:
Ok, good point. Let's consider this then, if we'll find it decongests the pages enough, we could do it on all of them.

Reply:
I decided to try this extraction out in this PR 😊 Please check.

Reply:
Cool, and you chose a better name :D